spirv_hlsl.cpp

```diff
@@ -1508,7 +1512,45 @@ void CompilerHLSL::emit_buffer_block(const SPIRVariable &var)
 
 void CompilerHLSL::emit_push_constant_block(const SPIRVariable &var)
 {
-	emit_buffer_block(var);
+	std::cout << options.root_constants_layout.size() << std::endl;
+	if (options.root_constants_layout.empty())
+	{
+		emit_buffer_block(var);
+		return;
+	}
+	else
+	{
+		for(auto const& layout : options.root_constants_layout)
+		{
+			auto &type = get<SPIRType>(var.basetype);
+
+			flattened_structs.insert(var.self);
+			type.member_name_cache.clear();
+			add_resource_name(var.self);
```

Comment:
If multiple root constant layout ranges are used, the name of var will get out of sync, probably best to call the cbuffer something fixed like SPIRV_Cross_RootConstants(Number) or something.

Reply:
I used a numbering scheme similar to the one proposed, just wondering if just renaming the cbuffer is fine as currently done by me?
It will emit something like
```
cbuffer SPIRV_CROSS_RootConstant_u_foo : register(b0, space0)
{
    float foo_a : packoffset(c0);
};
cbuffer SPIRV_CROSS_RootConstant_u_foo_1 : register(b1, space0)
{
    float3 foo_b : packoffset(c0.w); // <- TODO: offset
};
```